tracer/test/Datadog.Trace.Tests/Agent/MessagePack/SpanMessagePackFormatterTests.cs

```diff
@@ -208,6 +208,156 @@ public void SpanLink_Tag_Serialization()
         }
     }
 
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    public void SpanEvent_Tag_Serialization(bool nativeSpanEventsEnabled)
+    {
+        var discoveryService = new DiscoveryServiceMock();
+        var mockApi = new MockApi();
+        var settings = TracerSettings.Create(new());
+        var agentWriter = new AgentWriter(mockApi, statsAggregator: null, statsd: null, automaticFlush: false);
+        var tracer = new Tracer(settings, agentWriter, sampler: null, scopeManager: null, statsd: null, NullTelemetryController.Instance, discoveryService);
+
+        tracer.TracerManager.Start();
+        discoveryService.TriggerChange(spanEvents: nativeSpanEventsEnabled);
```

Comment:
Can you also test what happens before we consume a change to the discovery service? I expect that before we receive the discovery service notification that span events is allowed that we will always encode span events as a span tag

Reply:
Updated the test to not call the trigger whatsoever in one case and then check the JSON was added.